temporal-testing-junit5/src/main/java/io/temporal/testing/TestActivityExtension.java

```diff
@@ -0,0 +1,177 @@
+      for (POJOActivityInterfaceMetadata activityInterface : metadata.getActivityInterfaces()) {
+        supportedParameterTypes.add(activityInterface.getInterfaceClass());
+      }
+    }
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  @Override
+  public boolean supportsParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+
+    if (parameterContext.getParameter().getDeclaringExecutable() instanceof Constructor) {
+      // Constructor injection is not supported
+      return false;
+    }
+
+    Class<?> parameterType = parameterContext.getParameter().getType();
+    return supportedParameterTypes.contains(parameterType);
+  }
+
+  @Override
+  public Object resolveParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+
+    TestActivityEnvironment testEnvironment = getTestEnvironment(extensionContext);
+
+    Class<?> parameterType = parameterContext.getParameter().getType();
+    if (parameterType == TestActivityEnvironment.class) {
+      return testEnvironment;
+    } else {
+      // Activity stub
+      return testEnvironment.newActivityStub(parameterType);
+    }
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext context) throws Exception {
+    TestActivityEnvironment testEnvironment =
+        TestActivityEnvironment.newInstance(testEnvironmentOptions);
+    testEnvironment.registerActivitiesImplementations(activityImplementations);
+
+    setTestEnvironment(context, testEnvironment);
+  }
+
+  @Override
+  public void afterEach(ExtensionContext context) throws Exception {
+    TestActivityEnvironment testEnvironment = getTestEnvironment(context);
+    testEnvironment.close();
+  }
+
+  private TestActivityEnvironment getTestEnvironment(ExtensionContext context) {
+    return getStore(context).get(TEST_ENVIRONMENT_KEY, TestActivityEnvironment.class);
+  }
+
+  private void setTestEnvironment(
+      ExtensionContext context, TestActivityEnvironment testEnvironment) {
+    getStore(context).put(TEST_ENVIRONMENT_KEY, testEnvironment);
+  }
+
+  private ExtensionContext.Store getStore(ExtensionContext context) {
+    Namespace namespace =
+        Namespace.create(TestActivityExtension.class, context.getRequiredTestMethod());
+    return context.getStore(namespace);
+  }
+
+  public static class Builder {
+
+    private static final Object[] NO_ACTIVITIES = new Object[0];
+
+    private TestEnvironmentOptions testEnvironmentOptions =
+        TestEnvironmentOptions.getDefaultInstance();
+    private Object[] activityImplementations = NO_ACTIVITIES;
+
+    private Builder() {}
+
+    /** @see TestActivityEnvironment#newInstance(TestEnvironmentOptions) */
+    public Builder setTestEnvironmentOptions(TestEnvironmentOptions testEnvironmentOptions) {
+      this.testEnvironmentOptions = testEnvironmentOptions;
+      return this;
+    }
```

Comment:
I'm not really sure what `TestEnvironmentOptions` options might make sense for `TestActivityEnvironment`.